ddprof-exporter/src/multipart.rs

```diff
@@ -0,0 +1,741 @@
+use std::borrow::Cow;
```

Comment:
were you able to take this from reqwest ?
Asking to understand how much effort we should spend reviewing this 😅

Reply:
Yeah, all this part is taken from reqwest, with few types adjustments because it used `reqwest::Body`, which I switched for `hyper::Body`.

I can probably separate my additions, from the files I copied to make it easier to review.

Reply:
I think it is fine as long as we are aware of it. Leaving the comment for future reviewers. Thank you !

Reply:
Was this looked at and deemed unsuitable? https://github.com/ferristseng/rust-multipart-rfc7578

Reply:
Actually no, I somehow missed it when looking for multipart libraires for hyper (now it's the first result on google??).
The code looks ok and it works with hyper 0.14 which wasn't the case of other libraires I look into, so I can try it and check if it's suitable.

Reply:
I checked it and it doesn't work. [rust-multipart-rfc7578](https://github.com/ferristseng/rust-multipart-rfc7578) writes the final boundary of the form after each part, which is clearly a bug.

So for instance if multiple parts are written in the body, this lib emit the following body

```
content-type: multipart/form-data; boundary=abc

--abc
content-type: text/plain
content-disposition: form-data; name="version"

3
--abc--
--abc
content-type: text/plain
content-disposition: form-data; name="start"

2022-02-07T15:59:53Z
--abc--
--abc
content-type: text/plain
content-disposition: form-data; name="end"

2023-11-11T16:00:00Z
--abc--
```

When it should be

```
content-type: multipart/form-data; boundary=abc

--abc
content-type: text/plain
content-disposition: form-data; name="version"

3
--abc
content-type: text/plain
content-disposition: form-data; name="start"

2022-02-07T15:59:53Z
--abc
content-type: text/plain
content-disposition: form-data; name="end"

2023-11-11T16:00:00Z
--abc--
```

Reply:
I would personally prefer that we contribute upstream and use something else than maintain it directly ourselves. At the very least, can you file an issue?

Reply:
I agree that we don't want to maintain this ourselves.
Reviewing the code to find the root of the issue, I found a bunch of other problems, but the fixes weren't too bad, so I opened a couple PR on the repo .

https://github.com/ferristseng/rust-multipart-rfc7578/pull/30
https://github.com/ferristseng/rust-multipart-rfc7578/pull/31

The maintainer seems responsive, and the first one got merged and released very quickly.
In the meantime, I added a dependency to my fork in the `Cargo.toml` to verify that the ruby integration tests were still passing.